packages/diagrams/frontend/sirius-components-diagrams/src/renderer/palette/usePaletteContents.tsx

```diff
@@ -76,21 +80,13 @@ const isDiagramDescription = (
   representationDescription: GQLRepresentationDescription
 ): representationDescription is GQLDiagramDescription => representationDescription.__typename === 'DiagramDescription';
 
-export const usePaletteContents = (diagramElementIds: string[]): UsePaletteContentValue => {
-  const { diagramId, editingContextId } = useContext<DiagramContextValue>(DiagramContext);
+export const usePaletteContents = (): UsePaletteContentValue => {
   const { addErrorMessage } = useMultiToast();
 
-  const {
-    data: paletteData,
-    loading,
-    error: paletteError,
-  } = useQuery<GQLGetToolSectionsData, GQLGetToolSectionsVariables>(getPaletteQuery, {
-    variables: {
-      editingContextId,
-      diagramId,
-      diagramElementIds,
-    },
-  });
+  const [getPaletteContents, { data: paletteData, loading, error: paletteError }] = useLazyQuery<
```

Comment:
The query has to be lazy now so we can call it on demand when a key binding is detected.